task/08-objects-tasks.js

```diff
@@ -105,36 +111,75 @@ function fromJSON(proto, json) {
  *
  *  For more examples see unit tests.
  */
+class CssSelector {
+
+    constructor(value) {
+        this.selector = '';
+        this.type = 0;
+        this.uniqueError = "Element, id and pseudo-element should not occur more then one time inside the selector";
+        this.orderError = "Selector parts should be arranged in the following order: element, id, class, attribute, pseudo-class, pseudo-element";
+    }
+
+    element(value) {
+        this.checkType(1);
+        this.selector += value;
+        return this;
+    }
+
+    id(value) {
+        this.checkType(2);
+        this.selector += `#${value}`;
+        return this;
+    }
+
+    class(value) {
+        this.checkType(3);
+        this.selector += `.${value}`;
+        return this;
+    }
+
+    attr(value) {
+        this.checkType(4);
+        this.selector += `[${value}]`;
+        return this;
+    }
+
+    pseudoClass(value) {
+        this.checkType(5);
+        this.selector += `:${value}`;
+        return this;
+    }
+
+    pseudoElement(value) {
+        this.checkType(6);
+        this.selector += `::${value}`;
+        return this;
+    }
+
+    combine(selector1, combinator, selector2) {
+        this.selector = `${selector1.stringify()} ${combinator} ${selector2.stringify()}`;
+        return this;
+    }
+
+    stringify() {
+        return this.selector;
+    }
+
+    checkType(type) {
+        if (type < this.type) throw this.orderError;
```

Comment:
Do test allow to use `new Error` class? If answer is "yes", you have to use `new Error` class instead of `string`.